### ISSUE TITLE
Unused uses and intersection types

### ISSUE THREAD

**Issue by Stadly** [slevomat/coding-standard#594]
[`SlevomatCodingStandard.Namespaces.UnusedUses`](https://github.com/slevomat/coding-standard#slevomatcodingstandardnamespacesunuseduses-) does not consider [intersection types](https://medium.com/@ondrejmirtes/union-types-vs-intersection-types-fd44a8eacbb) when searching annotations.

For instance the below code erroneously reports that `Type PHPUnit\Framework\MockObject\MockObject is not used in this file.`.

```php
use PHPUnit\Framework\MockObject\MockObject;

class Foo
{
    /**
     * @var MockObject&Bar
     */
    private $mockedObject;
}
```

**Comment by kukulich** [comment]
Duplicate of https://github.com/slevomat/coding-standard/issues/251
